### DISCUSSION TITLE
Unable to delete the scaled object deployed in Azure AKS and Namespace in terminating state

### DISCUSSION THREAD

**Discussion by nidiculageorge** [kedacore/keda#5131] [Q&A / Need Help] [answered]
### Report

Unable to delete the scaled object deployed in Azure AKS and Namespace in terminating state

![image](https://github.com/kedacore/keda/assets/43850474/9244b07b-ae10-4617-81e8-d8935309c6bb)

### Expected Behavior

As we tried deleting the namespace its in terminating state because of the above scaled object in that namespace

### Actual Behavior

Namesapce should be deleted after the scaled object is deleted

### Steps to Reproduce the Problem

1.
2.
3.

### Logs from KEDA operator

```
example
```

### KEDA Version

2.11.1

### Kubernetes Version

1.25

### Platform

Microsoft Azure

**Comment by nidiculageorge** [comment]
I tried removing the finalizer but still I am unable to remove the Scaled object the finalizer is not getting removed from the scaled object

![image](https://github.com/kedacore/keda/assets/43850474/e1e54619-a7ad-443d-9f2e-1abf7625f6c1)

**Comment by SpiritZhou** [comment]
Could you provide the events of scaledobject or logs of keda-operator?

**Comment by zroubalik** [comment]
Do you have KEDA operator up and running?

**Comment by nidiculageorge** [comment]
@zroubalik  we have keda operator running running in the cluster.But other applications are also using this same Keda

![image](https://github.com/kedacore/keda/assets/43850474/fc1464f9-59d7-46f1-adce-a71e42ea5b21)

**Comment by zroubalik** [comment]
So try to remove these scaledobjects first, then namespace

**Comment by nidiculageorge** [comment]
@zroubalik  this is keda namespace .Could you please clarify.

Or do we need to delete this 

![image](https://github.com/kedacore/keda/assets/43850474/4be22119-527a-46b5-9808-2345ea34252f)

**Comment by zroubalik** [comment]
just delete all scaledobjects in the namespace you are about to remote:

```bash
k delete scaledobject --all -n YOUR_NAMESPACE
```

And also check keda-operator pod logs to see if there are any errors or anything

**Comment by nidiculageorge** [comment]
@zroubalik I tried deleting the same and is stuck like this

![image](https://github.com/kedacore/keda/assets/43850474/1eecf298-ce84-4302-a53c-9d6d669d8a38)

**Comment by nidiculageorge** [comment]
@zroubalik I could see some errors in the keda operator log related to this namespace which we are trying to delete

2023-10-19T03:28:51Z    ERROR   scale_handler   error getting metric for scaler {"scaledObject.Namespace": "autozoning-live-ibr-34", "scaledObject.Name": "recognitionicr-autoscaler", "scaler": "azureServiceBusScaler", "error": "scaler with id 2 not found, len = 0, cache has been probably already invalidated"}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).GetScaledObjectMetrics
        /workspace/pkg/scaling/scale_handler.go:485
github.com/kedacore/keda/v2/pkg/metricsservice.(*GrpcServer).GetMetrics
        /workspace/pkg/metricsservice/server.go:48
github.com/kedacore/keda/v2/pkg/metricsservice/api._MetricsService_GetMetrics_Handler
        /workspace/pkg/metricsservice/api/metrics_grpc.pb.go:98
google.golang.org/grpc.(*Server).processUnaryRPC
        /workspace/vendor/google.golang.org/grpc/server.go:1336
google.golang.org/grpc.(*Server).handleStream
        /workspace/vendor/google.golang.org/grpc/server.go:1704
google.golang.org/grpc.(*Server).serveStreams.func1.2
        /workspace/vendor/google.golang.org/grpc/server.go:965
2023-10-19T03:28:51Z    ERROR   scale_handler   error getting metric spec for the scaler        {"scaledObject.Namespace": "autozoning-live-ibr-34", "scaledObject.Name": "recognitionicr-autoscaler", "scaler": "azureServiceBusScaler", "error": "scaler with id 3 not found. Len = 0"}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).GetScaledObjectMetrics
        /workspace/pkg/scaling/scale_handler.go:446
github.com/kedacore/keda/v2/pkg/metricsservice.(*GrpcServer).GetMetrics
        /workspace/pkg/metricsservice/server.go:48
github.com/kedacore/keda/v2/pkg/metricsservice/api._MetricsService_GetMetrics_Handler
        /workspace/pkg/metricsservice/api/metrics_grpc.pb.go:98
google.golang.org/grpc.(*Server).processUnaryRPC
        /workspace/vendor/google.golang.org/grpc/server.go:1336
google.golang.org/grpc.(*Server).handleStream
        /workspace/vendor/google.golang.org/grpc/server.go:1704
google.golang.org/grpc.(*Server).serveStreams.func1.2
        /workspace/vendor/google.golang.org/grpc/server.go:965

**Comment by nidiculageorge** [comment]
@zroubalik any update

**Comment by nidiculageorge** [comment]
Hi Team,

Is there any update on the issue.

**Reply by JorTurFer** [reply, marked as answer]
To delete the ScaledObject, KEDA operator must be alive to remove the finalizer. Is the operator alive when you delete the ScaledObject?
You can manually edit the ScaledObject and remove the finalizer to remove the ScaledObject, despite the operator pod is down

**Comment by JuanRamino** [comment]
Manually remove the finalizer from scaledobjects is not a solution.
We are experiencing a long time wait when deleting a namespace caused by scaledobjects.
The operator is always alive when this happen.
Sometimes it manages to delete the namespace after a few minutes, others after 10 minutes but it is never a quick process
Why this happen?

**Reply by JorTurFer** [reply]
Do you see any error related with throttling in operator logs?
